src/containers/layouts/ecommerce/productsList/productsList.component.tsx

```diff
@@ -0,0 +1,175 @@
+  ProductProps,
+  Product,
+} from '@src/components/ecommerce';
+
+interface ComponentProps {
+  products: ProductModel[];
+  onAddToBucketPress: (product: ProductModel) => void;
+}
+
+interface State {
+  selectedIndex: number;
+  currentProducts: ProductModel[];
+}
+
+export type ProductsListProps = ThemedComponentProps & ComponentProps;
+
+class ProductsListComponent extends React.Component<ProductsListProps, State> {
+
+  public state: State = {
+    selectedIndex: 0,
+    currentProducts: this.props.products,
+  };
+
+  private extractKey = (item: string, index: number): string => {
+    return `${item}-${index}`;
+  };
+
+  private onSelectTab = (selectedIndex: number, category: string): void => {
+    const products: ProductModel[] = this.filterProducts(category);
+    this.setState({
+      selectedIndex: selectedIndex,
+      currentProducts: products,
+    });
+  };
+
+  private onAddToBucketPress = (product: ProductModel): void => {
+    this.props.onAddToBucketPress(product);
+  };
+
+  private getComponentsTabs = (): string[] => {
+    const { products } = this.props;
+    const tabs: string[] = products.map((item: ProductModel) => item.type);
+
+    return ['ALL', ...new Set(tabs)];
+  };
+
+  private filterProducts = (criteria: string): ProductModel[] => {
+    const { products } = this.props;
+
+    if (criteria === 'ALL') {
+      return products;
+    } else {
+      return products.filter((item: ProductModel) => item.type === criteria);
+    }
+  };
+
+  private renderTab = (info: ListRenderItemInfo<string>): React.ReactElement<TabProps> => {
+    const { themedStyle } = this.props;
+    const isSelected: boolean = info.index === this.state.selectedIndex;
+
+    return (
+      <TouchableOpacity
+        onPress={() => this.onSelectTab(info.index, info.item)}
+        style={themedStyle.tabContainer}>
+        <Text style={themedStyle.tabLabel(isSelected)}>{info.item.toUpperCase()}</Text>
+        {isSelected && <View style={themedStyle.tabIndicator}/>}
+      </TouchableOpacity>
+    );
+  };
+
+  private renderProduct = (info: ListRenderItemInfo<ProductModel>): React.ReactElement<ProductProps> => {
+    const { themedStyle } = this.props;
+
+    return (
+      <Product
+        style={themedStyle.productItem}
+        product={info.item}
+        index={info.index}
+        onAddToBucket={this.onAddToBucketPress}
+      />
+    );
+  };
+
+  private renderProducts = (): React.ReactElement<ViewProps> => {
+    const { themedStyle } = this.props;
+
+    return (
+      <View style={themedStyle.productsList}>
+        <List
+          contentContainerStyle={themedStyle.productsListContent}
+          data={this.state.currentProducts}
+          renderItem={this.renderProduct}
+          numColumns={2}/>
+      </View>
+    );
+  };
+
+  public render(): React.ReactNode {
+    const { themedStyle } = this.props;
+    const tabs: string[] = this.getComponentsTabs();
+
+    return (
+      <View style={themedStyle.container}>
```

Comment:
Could this be achieved with `contentContainerStyle` prop?

Reply:
Refactored